server/manager/src/main/java/org/apache/accumulo/master/upgrade/Upgrader9to10.java

```diff
@@ -201,6 +204,39 @@ private void upgradeRootTabletMetadata(ServerContext ctx) {
     delete(ctx, ZROOT_TABLET_PATH);
   }
 
+  @SuppressWarnings("removal")
+  @Deprecated(since = "2.1.0", forRemoval = true)
+  private void renameMasterProps(ServerContext ctx) {
+    try {
+      // Figure out which props with the "master." prefix are set only in Zookeeper. If
+      // a property exists in the config and is either not set in the site config, or has
+      // a different value from the site config, then the property is set in zookeeper
+      // and we want to upgrade.
+      HashMap<String,String> configProps = new HashMap<>();
+      HashMap<String,String> siteProps = new HashMap<>();
+      ctx.getConfiguration().getProperties(configProps,
+          p -> p.startsWith(DeprecatedPropertyUtil.MasterPropertyRenamer.MASTER_PREFIX));
+      ctx.getSiteConfiguration().getProperties(siteProps,
+          p -> p.startsWith(DeprecatedPropertyUtil.MasterPropertyRenamer.MASTER_PREFIX));
+      configProps.entrySet().removeIf(
+          e -> siteProps.containsKey(e.getKey()) && e.getValue().equals(siteProps.get(e.getKey())));
```

Comment:
Seems like if `master.prop1=v1` is set in site and ZK, then it will not be renamed in ZK?  If so, why not rename it in ZK?

Reply:
I was trying to list only properties that are set in ZK. I couldn't find a better way to determine that, and modeled this after code in [ConfigCommand](https://github.com/apache/accumulo/blob/main/shell/src/main/java/org/apache/accumulo/shell/commands/ConfigCommand.java).
In the case you mention, the ZK property is superfluous and really should be removed rather than renamed if it matches what's in site already, but I'm not doing that either.
Is there another utility for retrieving ZK properties while ignoring the contents of the site file, command-line and env properties? Maybe I need to write something if not.

Reply:
I missed that configProps was a union of the site config and ZK config, I see what you are trying to do now.

> Is there another utility for retrieving ZK properties while ignoring the contents of the site file

Looking around in the code, one way might be to create a ZooConfiguration object directly with an empty parent.

Reply:
I tried that, and it looks a little cleaner (and catches the edge case). In the next commit...